packages/shared/table/selectable-table.tsx

```diff
@@ -0,0 +1,154 @@
+import * as React from 'react';
+import { K8sResourceCommon } from '@openshift-console/dynamic-plugin-sdk';
+import { Bullseye } from '@patternfly/react-core';
+import {
+  SortByDirection,
+  TableComposable,
+  Tbody,
+  Td,
+  Th,
+  ThProps,
+  Thead,
+  Tr,
+} from '@patternfly/react-table';
+import { useSelectList } from '../hooks/select-list';
+import { useSortList } from '../hooks/sort-list';
+import { getUID } from '../selectors';
+import { useCustomTranslation } from '../useCustomTranslationHook';
+
+const isRowSelectable = <T extends K8sResourceCommon>(row: T) =>
+  !row?.metadata?.deletionTimestamp;
+
+const areAllRowsSelected = <T extends K8sResourceCommon>(
+  selectableRows: T[],
+  selectedRows: T[]
+) => {
+  const selectedRowIds = selectedRows?.map((selectedRow) =>
+    getUID(selectedRow)
```

Comment:
```suggestion
  const selectedRowIds = selectedRows?.map(getUID)
```
nit: you could do something like this as well.